### PR TITLE
fix genericsetup export with complex properties 

### LINKED ISSUE #50: Genericsetup exportimport doesn't handle dictionary field types
for example a rule with a schema with ```value_type=schema.Dict(value_type=schema.Field(),
                               key_type=schema.TextLine()),``` is going to exported as text and then will fail when imported.

The bug is here - https://github.com/plone/plone.app.contentrules/blob/master/plone/app/contentrules/exportimport/rules.py#L123.

I'm not sure but there should be some common code to do XML marshalling of schema defined values right?

### PR DESCRIPTION
**Pull request by djay**
fixes #50

### CODE REVIEW

**Comment by djay** [comment]
@jenkins-plone-org please run jobs

**Comment by djay** [comment]
problem with the test is that Sets don't get sorted on export - https://github.com/plone/plone.supermodel/issues/29. The previous code was wrong, in that it sorted lists as well :(

**Comment by djay** [comment]
@jenkins-plone-org please run jobs

File: plone/app/contentrules/exportimport/rules.py [hunk 1]
```diff
@@ -113,54 +100,16 @@ def export_field(self, doc, field):
...
-                child.appendChild(doc.createTextNode(six.text_type(value)))
-
+        
+        node = valueToElement(field, value)
+        if node.text:
+            child.appendChild(doc.createTextNode(six.text_type(node.text)))
+        # Assumes there are not other text nodes and we can throw away the parent node    
+        for node in node.iterchildren():
+            xml = etree.tostring(node, encoding="utf8")
+            child.appendChild(minidom.parseString(xml).firstChild)
```
Comment (jensens) [inline plone/app/contentrules/exportimport/rules.py]: I am confused? Why ``minidom`` here if we have `lxml` already? If it is needed: at least a comment would help here

Comment (djay) [reply plone/app/contentrules/exportimport/rules.py]: Because export import code is using minidom already and rewriting all of it is not worth it.

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by djay** [comment]
@jenkins-plone-org please run jobs

**Comment by djay** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**